### PR TITLE
fix(feeds): RSS improve torrent size parsing

### PR DESCRIPTION
**Pull request by ghoto**
When some RSS feeds have `Enclosure->Length` fixed size, autobrr fails to retrieve the size that's included in `Custom->Size`.

The previous logic used `custom->size` when `enclosure->length` was 0. But as stated in a previous issue, this `enclosure->length` can be a fixed length unrelated to the torrent. Therefore this value can be 0, fixed length, or the real one.

This modification will pick `custom->size` whenever is defined and it's bigger than `enclosures->length`.

https://github.com/autobrr/autobrr/issues/636

### CODE REVIEW

File: internal/feed/rss.go [hunk 1]
```diff
@@ -12,6 +12,7 @@ import (
 	"github.com/autobrr/autobrr/internal/release"
 	"github.com/autobrr/autobrr/pkg/errors"
 
+	"github.com/dustin/go-humanize"
```
Comment (KyleSanderson) [inline internal/feed/rss.go:15]: Can you revert this?

File: internal/feed/rss.go [hunk 2]
```diff
@@ -154,10 +155,12 @@ func (j *RSSJob) processItem(item *gofeed.Item) *domain.Release {
...
 	}
 
-	if rls.Size == 0 {
-		// parse size bytes string
-		if sz, ok := item.Custom["size"]; ok {
-			rls.ParseSizeBytesString(sz)
+	
+	// When custom->size and enclosures->size differ, pick the largest one.
+	if size, ok := item.Custom["size"]; ok {
+		s, err := humanize.ParseBytes(size)
```
Comment (KyleSanderson) [inline internal/feed/rss.go:161]: Ditto.

**Review by KyleSanderson — CHANGES_REQUESTED** [review]
Thank you very much for the contribution. Can you please revert the module changes (if there's a problem with the existing function, that should be fixed instead).

**Comment by ghoto** [comment]
What do you mean with the existing function?

**Comment by ghoto** [comment]
@KyleSanderson I moved the logic to `ParseSizeBytesString`. I fixed another potential bug on the function that would reset the size of the Release to 0 if the string passed to the function was malformed.

**Review by KyleSanderson — APPROVED** [review]

**Review by zze0s — COMMENTED** [review]
If we are making changes then might be good to check for not only enclosure size difference but quite a bit of feeds set the enclosure size to a static `39399` on all items.

While it could happen that something actually have that exact size, it's imo unlikely.

**Review by zze0s — APPROVED** [review]

**Pull request merged**